src/analyses/base.ml

```diff
@@ -1261,9 +1261,29 @@ struct
   *)
   (* TODO: deduplicate https://github.com/goblint/analyzer/pull/1297#discussion_r1477804502 *)
   let rec exp_may_signed_overflow ctx exp =
+    let binop = (GobOption.map2 Z.div )in
```

Comment:
What's this?

Reply:
Right now this `binop` is in the entire scope of `exp_may_signed_overflow` and it's specifically for `Z.div` (but the name makes no mention of this.

Maybe this should be pushed down into `checkDiv` for which it was extracted.

Reply:
You are completely right about this - moved it deeper into checkDiv now